LICENSE.txt

```diff
@@ -1,5 +1,6 @@
 The MIT License (MIT)
 Copyright (c) 2012-2021: Timothy E. Holy, Konrad Hinsen, Tom Short, and Simon Kornblith.
+Copyright (c) 2021: Mark Kittisopikul and Howard Hughes Medical Institute.
```

Comment:
```suggestion
```

Changing the existing license without consulting and receiving approval from all the existing copyright holders  is not permitted.

Reply:
We are not changing the license. We are simply adding an additional copyright notice as permitted by the MIT license.

Reply:
@timholy, I would appreciate your opinion here.

Reply:
I've reverted the change to LICENSE.txt in cc92c6b because the copyright notice in the source comments are sufficient:
https://github.com/JuliaIO/HDF5.jl/pull/875/files#diff-32e7a0ac02390300f5785e4de5d5490a9f47acce10c9de854a39f123794f2584R10-R12

However, the situation is confusing since those comments refer to LICENSE.txt and the copyright line does not appear to include the attribution and copyright assignment in those comments. I am not a lawyer. That said my understanding is the copyright for the contents of this repository is still retained by all contributors to the repository under US law and in other countries which are signatories of the Berne Convention and not just the four individuals currently listed.

I think the best practice here would be to add or amend the line to include all contributors to HDF5.jl similar to the copyright notice used by Julia itself:
https://github.com/JuliaLang/julia/blob/master/LICENSE.md?plain=1#L3

This would follow the recommendation made in this Linux Foundation Blog:
https://www.linuxfoundation.org/blog/copyright-notices-in-open-source-software-projects/

Reply:
Do you mind clarifying exactly what your company's IP department has stated? My original concern/point of hesitation at such changes here was that it seemed obnoxious to add our name to the LICENSE based on a PR contribution (which I haven't seen before in the past with other repos I've been involved in).  But if you received legal recommendations otherwise, we can add it here and move on with this issue. 

Looking at https://github.com/JuliaIO/HDF5Plugins.jl/blob/main/LICENSE we should probably also then just append all the other licenses from the ported over module to the same file here at HDF5.jl, instead of keeping them in a separate license folder (my original recommendation).

Reply:
The position of Howard Hughes Medical Institute (HHMI) in this case is:
1) HHMI asserts copyright over the contributed code per the Intellectual Property Policy, of which I subject to as an employee at the Janelia Research Campus:
https://hhmicdn.blob.core.windows.net/policies/Intellectual-Property-Policy
2) HHMI licenses the contributed code the via the MIT license, in this case such that the terms of the licenses also apply to HHMI as a licensor.
3) HHMI prefers a 3-clause BSD License, https://www.janelia.org/node/47808, but has made an exception here to license under a MIT license.

The last paragraph is particularly important, although all clauses apply:
https://github.com/JuliaIO/HDF5.jl/blob/b349f57c9642577e493747be92331ec0b02b97e5/LICENSE.txt#L14-L18

There was some discussion of whether my port is a derivative work or not, but in an abundance of caution reproducing the licenses per the terms of those licenses should be sufficient. I think keeping the third party licenses in a separate folder is fine.

Reply:
So adding 
```
Copyright (c) 2021: Mark Kittisopikul and Howard Hughes Medical Institute.
```
should be sufficient. Do  you need the additional notices on all the files you modified (that seems excessive if it's already in the LICENSE file?)

Reply:
The above works; however, adding, "and contributors" in the license file should be fine as well. The comment in source is sufficient in conjunction with that addition.

As for notices you every file

For example, almost every source code file in Julia contains the following at the top:
> This file is a part of Julia. License is MIT: https://julialang.org/license

Reply:
Resolved by https://github.com/JuliaIO/HDF5.jl/pull/879